src/util.cc

```diff
@@ -461,6 +457,18 @@ std::string WideToUtf8(const WStringPiece& w_string) {
   if (size_needed == 0)
     Fatal("Failed conversion to UTF-8 string");
 
+  return size_needed;
+}
+
+std::string WideToUtf8(const WStringPiece& w_string) {
+  if (w_string.len_ == 0)
+    return string();
+
+  // Make sure that the string size can be casted to an int.
+  assert(!(w_string.size() > INT_MAX));
+
+  int size_needed = GetUtf8SizeNeeded(w_string);
+
   std::string u8_string(size_needed, '\0');
   int u8_size_needed = WideCharToMultiByte(
       CP_UTF8, 0, w_string.wstr_, static_cast<int>(w_string.size()),
```

Comment:
For the if check below

`if (u8_size_needed == 0 || (u8_size_needed > size_needed))`

wouldn't simply

`if(u8_size_needed != size_needed)`

cover all the possibilities?

Reply:
The return value of the conversion will return zero if it fails, but can also return the needed size in case the buffer is insufficient. The check covers both cases, even though the second one is highly unlikely.

Reply:
Right, that's my point.

If the returned value is not size_needed, then either

1) The conversion failed
2) The conversion took a different amount of space than the previous call to the same function said it would.

In all cases, the only time we want to proceed, is if the return value is exactly what the previous call to the function (to get the needed size) said it would be, and since we check the return value against zero when we call it the first time, we don't need a compound check.

Reply:
Sorry, missread the comment. Good point!

Reply:
Looks good.

However... please don't skip the opening and closing brackets on if statements.

That's how vulnerabilities like heartbleed happen.